flytekit/core/artifact.py

```diff
@@ -400,19 +493,29 @@ def embed_as_query(
         partition: Optional[str] = None,
         bind_to_time_partition: Optional[bool] = None,
         expr: Optional[str] = None,
+        op: Optional[Op] = None,
     ) -> art_id.ArtifactQuery:
         """
         This should only be called in the context of a Trigger
         :param partition: Can embed a time partition
         :param bind_to_time_partition: Set to true if you want to bind to a time partition
         :param expr: Only valid if there's a time partition.
+        :param op: If expr is given, then op is what to do with it.
         """
         # Find self in the list, raises ValueError if not there.
+        t = None
+        if expr and (partition or bind_to_time_partition):
+            if op == Op.MINUS:
+                t = art_id.TimeTransform(transform=expr, op=art_id.Operator.MINUS)
+            elif op == Op.PLUS:
+                t = art_id.TimeTransform(transform=expr, op=art_id.Operator.PLUS)
+            else:
+                raise ValueError(f"Unknown or missing time partition operation {op}")
```

Comment:
Drop the comment and simplify this:
```suggestion
        t = None
        if expr and (partition or bind_to_time_partition):
            t = art_id.TimeTransform(transform=expr, op=op)
```